addons/onboarding/models/onboarding_progress_step.py

```diff
@@ -10,18 +10,24 @@ class OnboardingProgressStep(models.Model):
     _description = 'Onboarding Progress Step Tracker'
     _rec_name = 'step_id'
 
-    progress_id = fields.Many2one(
-        'onboarding.progress', 'Related Onboarding Progress Tracker', required=True, ondelete='cascade')
+    progress_ids = fields.Many2many(
+        'onboarding.progress', 'onboarding_progress_onboarding_progress_step_rel',
+        'onboarding_progress_step_id', 'onboarding_progress_id',
+        string='Related Onboarding Progress Tracker', required=False)
```

Comment:
Why required=False ? As it is the default value, and people generally do not expect 2many relationships to be required.